lib-gphotos/client.go

```diff
@@ -57,73 +56,141 @@ func NewClient(oauthHTTPClient *http.Client, maybeToken ...*oauth2.Token) (*Clie
 }
 
 // GetUploadToken sends the media and returns the UploadToken.
-func (client *Client) GetUploadToken(r io.Reader, filename string) (token string, err error) {
-	req, err := http.NewRequest("POST", fmt.Sprintf("%s/%s/uploads", basePath, apiVersion), r)
+func (c *Client) GetUploadToken(r io.ReadSeeker, filename, uploadURL string, fileSize int64) (string, string, error) {
+	var offset int64
+
+	if uploadURL != "" {
```

Comment:
I think it would be cleaner if you break this code in two functions:

- one to deal with uploads previously started (first `if` block)
- one to deal with new uploads (second `if` block)

The code would be easier to follow and to test :sweat_smile:

Reply:
Agreed, I'll look into that.

Reply:
Turns out the first `if` block can reset the `uploadURL` to empty if `status` is not `active` in which case we enter the second if block.